cmd/soroban-cli/src/strval.rs

```diff
@@ -909,24 +929,43 @@ pub fn to_json(v: &ScVal) -> Result<Value, Error> {
             .to_string();
             Value::String(v)
         }
-        ScVal::U256(Uint256(inner)) => {
-            let (hi, lo) = inner.split_at(16);
-            Value::String(
-                ethnum::U256::from_words(
-                    u128::from_be_bytes(hi.try_into()?),
-                    u128::from_be_bytes(lo.try_into()?),
-                )
-                .to_string(),
-            )
+        ScVal::U256(u256parts) => {
+            let hi_bytes = [u256parts.hi_hi.to_be_bytes(), u256parts.hi_lo.to_be_bytes()].concat();
+            let lo_bytes = [u256parts.lo_hi.to_be_bytes(), u256parts.lo_lo.to_be_bytes()].concat();
+            let hi = u128::from_be_bytes(
+                hi_bytes
+                    .as_slice()
+                    .try_into()
+                    .map_err(|_| Error::InvalidValue(Some(ScType::U256)))?,
+            );
+            let lo = u128::from_be_bytes(
+                lo_bytes
+                    .as_slice()
+                    .try_into()
+                    .map_err(|_| Error::InvalidValue(Some(ScType::U256)))?,
+            );
+            Value::String(ethnum::U256::from_words(hi, lo).to_string())
```

Comment:
These methods can be really fiddly. It would be great to add at least a simple unit test for these conversions.

Reply:
I'll work on some basic tests.

Reply:
Yep. good enough for now.

Reply:
Cool, I will create a ticket